### DISCUSSION TITLE
How to use 'json-stringify' parser to format json files.

### DISCUSSION THREAD

**Discussion by KyrieLii** [biomejs/biome#3670] [Q&A]
Hi, I'm currently migrating projects from prettier to biome.
In prettier, json files will be formated by ['json-stringify'](https://prettier.io/docs/en/options.html#parser): 
```
["INFO" - 23:32:30] Prettier Options:
{
  "filepath": "xxxx.json",
  "parser": "json-stringify",
  "singleQuote": true,
  "trailingComma": "all",
  "arrowParens": "avoid"
}
```
But how to do it using biome? 🙏

And there are an issue with formatting json files:

There are multiple blank lines between some lines: 
```json
{
  "a": "a",


  "b": "b"
}
```
Will be formatted but there is still a blank line:
```json
{
  "a": "a",

  "b": "b"
}
```
Is there a way to remove all empty lines ?

**Comment by dyc3** [comment]
Biome has support for formatting JSON. 

```
biome format example.json
biome format --write example.json
```

**Reply by KyrieLii** [reply]
I know that json can be formatted, my question is whether the formatting rules can be prettier's json-stringnify.

**Comment by ematipico** [comment]
We don't plan to match different parsers. Biome uses its own parser and does formatting one way, consistently across languages.

**Reply by KyrieLii** [reply]
Ok, i got it. But is there a way (or plan) to customize this behavior ?
I don't want to both use biome to format .ts .tsx and use prettier to format .json 😭 I want to use biome only .

**Reply by Conaclos** [reply]
Personally, I am in favor of supporting an option to use a `JSON.stringify` formatting style because many people expect it for the configuration file. To be honest, I don't even understand why it's not the default in Prettier. Prettier uses this style for some files such as `package.json`. We will need more discussion in the team to come to an agreement.

**Reply by KyrieLii** [reply]
@Conaclos TFYI ! looking forward to it!

**Reply by dyc3** [reply]
> Prettier uses this style for some files such as `package.json`

I'm pretty sure this is because package manager tools (I know `yarn` does this for sure) overwrite `package.json` even when no changes are necessary. For me, this resulted in my format CI job failing, because it would:
1. Run `yarn install`
2. yarn would overwrite `package.json` with new formatting
3. biome would see the change and fail the job.
I ended up spending way too much time trying to figure out what the problem was.

To fix this on one of my repos, I had to add this override to match the output of `JSON.stringify`.
```json
{
	"include": ["**/package.json"],
	"formatter": {
		"lineWidth": 40
	}
},
```

**Reply by KyrieLii** [reply]
@dyc3 I also encountered a similar problem using changesets, it will update package.json after publish , and biome fail too.
So I using both biome and prettier(for *.json) now ...
